Userland/DevTools/HackStudio/Dialogs/Commit/GitCommitDialog.cpp

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2021, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "GitCommitDialog.h"
+#include <DevTools/HackStudio/Dialogs/Commit/GitCommitDialogGML.h>
+
+namespace HackStudio {
+
+GitCommitDialog::GitCommitDialog(GUI::Window* parent)
+    : Dialog(parent)
+{
+    resize(400, 260);
+    center_within(*parent);
+    set_resizable(false);
+    set_modal(true);
+    set_title("Commit");
+    set_icon(parent->icon());
+
+    auto& widget = set_main_widget<GUI::Widget>();
+    widget.load_from_gml(git_commit_dialog_gml);
+
+    m_message_editor = *widget.find_descendant_of_type_named<GUI::TextEditor>("message_editor");
+    m_cancel_button = *widget.find_descendant_of_type_named<GUI::Button>("cancel_button");
+    m_commit_button = *widget.find_descendant_of_type_named<GUI::Button>("commit_button");
```

Comment:
Please use `widget->` instead of `*widget.` here. There's no need to confuse the reader :^). Alternatively, you can dereference the return value of `set_main_widget` if you're confident it won't be null. I can't speak to the specific details of LibGUI there, unfortunately.
```suggestion
    m_message_editor = widget->find_descendant_of_type_named<GUI::TextEditor>("message_editor");
    m_cancel_button = widget->find_descendant_of_type_named<GUI::Button>("cancel_button");
    m_commit_button = widget->find_descendant_of_type_named<GUI::Button>("commit_button");
```

Reply:
Yeah I'm not exactly sure. It also seems that it requires ``.`` and not ``->``. 
![image](https://user-images.githubusercontent.com/71222289/126876720-d5a37d6e-9e6d-49eb-903a-8d5ecca8bbc0.png)
![image](https://user-images.githubusercontent.com/71222289/126876735-62eff319-f364-47f0-9592-01bc11f0fa6c.png)